pubtools/_quay/push_docker.py

```diff
@@ -731,7 +727,28 @@ def run(self) -> None:
                 failed = True
 
         # Remove old signatures
-        outdated_manifests = self.get_outdated_manifests(backup_tags)
+        # run generate backup mapping again to fetch new digests of pushed containers
+        backup_tags2, _ = self.generate_backup_mapping(docker_push_items)
+        # if new backup tag has differnet digest, it means it was overwritten during the push
+        # and old signature should be removed. If the digest is the same it means, same item
+        # was just repushed
+        outdated_tags = {}
+        backup_tags2_shared = {}
+
+        # Backup tags can contain new tags which were orignally rollback_tags
+        # limit the comparision for outdated manifests to original backup_tags only
+        for bt2 in backup_tags2.items():
+            if bt2[0] in backup_tags:
+                backup_tags2_shared[bt2[0]] = bt2[1]
+
+        for bt1, bt2 in zip(
```

Comment:
Are the tags guaranteed to be in the same order? Wouldn't it be simpler to only iterate backup_tags1 and check for their existence in backup_tags2 and compare the values?

Reply:
That's why it's iterating over backup_tags2_shared which has only tags which are both in backup_tags and backup_tags2

Reply:
Your wording implies that backup_tags2_shared can contain fewer entries than backup_tags. Can that happen? If so, we cannot rely on  the values to correspond with each other, right?